android/src/main/java/com/amplitude/android/internal/gestures/AutocaptureGestureListener.kt

```diff
@@ -62,17 +64,133 @@ class AutocaptureGestureListener(
         distanceX: Float,
         distanceY: Float,
     ): Boolean {
+        val decorView = ensureWindowDecorView("onScroll") ?: return false
+        if (e1 == null || scrollState.type != MoveGestureType.Unknown) {
+            return false
+        }
+
+        val target: ViewTarget =
+            decorView.findTarget(
+                Pair(e1.x, e1.y),
+                viewTargetLocators,
+                ViewTarget.Type.Scrollable,
+                logger,
+            ) ?: logger.warn("Unable to find scroll target. No event captured.").let { return false }
+
+        scrollState.setTarget(target)
+        scrollState.type = MoveGestureType.Scroll
         return false
     }
 
-    override fun onLongPress(e: MotionEvent) {}
+    fun onUp(e: MotionEvent) {
+        ensureWindowDecorView("onUp") ?: return
+
+        val scrollTarget = scrollState.viewTarget ?: return
+        if (scrollState.type == MoveGestureType.Unknown) {
+            logger.warn("Unable to define scroll type. No event captured.")
+            return
+        }
+
+        val direction = scrollState.calculateDirection(e)
+        val eventType =
+            if (scrollState.type == MoveGestureType.Scroll) {
+                ELEMENT_SCROLLED
+            } else {
+                ELEMENT_SWIPED
+            }
+
+        capture(scrollTarget, eventType, mapOf(DIRECTION to direction))
+        scrollState.reset()
+    }
 
     override fun onFling(
         e1: MotionEvent?,
         e2: MotionEvent,
         velocityX: Float,
         velocityY: Float,
     ): Boolean {
+        scrollState.type = MoveGestureType.Swipe
         return false
     }
+
+    override fun onShowPress(e: MotionEvent) {}
+
+    override fun onLongPress(e: MotionEvent) {}
+
+    private fun ensureWindowDecorView(caller: String): View? {
+        val activity = activityRef.get()
+        val window = activity?.window
+        val decorView = window?.decorView
+
+        if (activity == null) {
+            logger.error("Activity is null in $caller")
+        } else if (window == null) {
+            logger.error("Window is null in $caller")
+        } else if (decorView == null) {
+            logger.error("DecorView is null in $caller")
+        }
+
+        return decorView
+    }
+
+    private fun capture(
+        target: ViewTarget,
+        eventType: String,
+        additionalProperties: Map<String, Any?> = emptyMap(),
+    ) {
+        mapOf(
+            ELEMENT_CLASS to target.className,
+            ELEMENT_RESOURCE to target.recourseName,
```

Comment:
nit:
```suggestion
            ELEMENT_RESOURCE to target.resourceName,
```

Reply:
😅
Fixed in https://github.com/amplitude/Amplitude-Kotlin/pull/201/commits/c10fc5dc8d3444c64f4534eae7e639f0d9d9dee3.